File: models/Role.js
const {DataTypes} = require('sequelize');
const {sequelize} = require('../config/database');

const Role = sequelize.define('Role', {
    role_id: {
        type: DataTypes.INTEGER,     // removed .UNSIGNED
        autoIncrement: true,
        primaryKey: true
    },
    name: {
        type: DataTypes.STRING(50),
        unique: true,
        allowNull: false
    }
}, {
    tableName: 'roles',
    schema: 'dbo',
    timestamps: false
});

module.exports = Role;

File: models/role.py
from sqlalchemy.orm import Mapped, mapped_column, relationship
from sqlalchemy import Integer, String
from .base import Base


class Role(Base):
    __tablename__ = "roles"
    __table_args__ = {"schema": "dbo"}

    role_id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)
    name: Mapped[str] = mapped_column(String(50), unique=True, nullable=False)

    Users = relationship("AuthUser", secondary="dbo.user_roles", back_populates="Roles")
